coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt

```diff
@@ -215,26 +173,123 @@ fun CoilImage(
             lastRequestedSize.value = requestSize
         }
 
-        if (painter == null) {
-            // If we don't have a result painter, we add a Box...
-            Box {
-                // If we don't have a result yet, we can show the loading content
-                // (if not null)
-                if (result == null && loading != null) {
-                    loading()
+        when (r) {
+            is SuccessResult -> image(r)
+            is ErrorResult -> if (error != null) error(r)
+            null -> {
+                // If we don't have a result painter, we add a Box...
+                Box {
```

Comment:
Do you need to add the `Box` still now that loading is a slot?

Reply:
Erm, probably not actually.